data_model/src/block.rs

```diff
@@ -88,7 +88,7 @@ mod model {
     )]
     #[display(fmt = "({header})")]
     #[allow(missing_docs)]
-    pub(crate) struct BlockPayload {
+    pub struct BlockPayload {
```

Comment:
can you explain yourself here? This shouldn't be public

Reply:
This warning appears when I run `cargo check -p iroha_swarm` (this is about `TransactionPayload`, but the cause is the same):

```
warning: type `transaction::model::TransactionPayload` is more private than the item `transaction::SignedTransaction::signatures`
   --> data_model/src/transaction.rs:297:5
    |
297 |     pub fn signatures(&self) -> &SignaturesOf<TransactionPayload> {
    |     ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ method `transaction::SignedTransaction::signatures` is reachable at visibility `pub`
    |
note: but type `transaction::model::TransactionPayload` is only usable at visibility `pub(crate)`
   --> data_model/src/transaction.rs:101:5
    |
101 |     pub(crate) struct TransactionPayload {
    |     ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
    = note: `#[warn(private_interfaces)]` on by default
```

However, it doesn't appear if I check `iroha`, or `iroha_core`, or `iroha_client` (targetly). I guess that is because they all by the end of the day use `iroha_data_model` with `transparent_api` enabled, while `iroha_swarm` uses it without any features.

I guess the temporary solution would be, first of all, to create an issue like "Fix all `transparent_api` usage across workspace" (you also mentioned `_transparent_api` here: https://github.com/hyperledger/iroha/pull/4382#discussion_r1565317258), but in scope of this PR either:

1. Return `pub(crate)` and silent the warning, temporarily
2. Enable `transparent_api` for `iroha_swarm`, temporarily

What do you think?

Reply:
check what @s8sato did [here](https://github.com/hyperledger/iroha/pull/4411#discussion_r1565275625)

if we don't want to leak private types here:
`pub fn signatures(&self) -> &SignaturesOf<TransactionPayload> {`
we can consider changing the function to:
`pub fn signatures(&self) -> &SignaturesOf<SignedTransaction> {`

> **I guess the temporary solution would be, first of all, to create an issue like "Fix all transparent_api usage across workspace" (you also mentioned _transparent_api here: https://github.com/hyperledger/iroha/pull/4382#discussion_r1565317258), but in scope of this PR either:**

there isn't much to fix. We just need to remove `transparent_api` from `test_network` because it's leaking into our integration tests and then client appears to have access to internals when in fact it doesn't. But that's too big of an effort so we have to live with it for now

Reply:
I tried to change it to `SignedTransaction` (and `SignedBlock`), but I need to transmute the type then. What is the proper way to do so?

Reply:
> if we don't want to leak private types here:
> pub fn signatures(&self) -> &SignaturesOf<TransactionPayload> {
> we can consider changing the function to:
> pub fn signatures(&self) -> &SignaturesOf<SignedTransaction> {

I don't think it's a good idea, as `SignedTransaction` is not scale-compatible with `TransactionPayload`. I guess we can make another versioned structure for transaction signatures and expose them as newtypes? Or we can expose the `TransactionPayload` itself..

Reply:
and why can't we just silence this lint instead of making any code changes?

Reply:
I would prefer that you open an issue, so we can merge this

Reply:
Reverted